keps/sig-node/3983-drop-in-configuration/README.md

```diff
@@ -0,0 +1,316 @@
+
+
+### Risks and Mitigations
+
+* Handling of zeroed fields
+    * It’s possible the configuration of the Kubelet does not handle not specified fields well. Special testing will need to be done for different types to define and ensure conformance of that behavior.
+* Handling of lists
+    * Contention could be found with how lists should be handled (append or overwrite, also see proposal). Consensus should be found and testing performed. 
+
+## Design Details
+
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+* Unit tests will be added, details to be added here
+
+##### Integration tests
+
+* :
+
+##### e2e tests
+
+* :
+
+### Graduation Criteria
+
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades are safe as far as Kubelet stability is concerned. It’s possible a vendor may ship vital pieces of configuration within a drop-in directory. If the Kubelet downgrades to a version that doesn’t support reading the drop-in directory, the kubelet will not recognize the "--config-dir" flag and risk failing. However, assuming that vendor left that the original `/etc/kubelet/kubelet.conf` is in a valid state, and the flag isn't specified, there should be no risk to the system. Any configuration that exists in a drop-in dir won't be applied, but that would not affect kubelet stability.
+
+
+### Version Skew Strategy
+
+All behavior change is encapsulated within the Kubelet, so there is no version skew possible within core Kubernetes. It is possible third party tools may attempt to utilize the Kubelet’s drop-in directory before the Kubelet is upgraded to support it, which would cause silent failures. It is the responsibility of these third party tools to ensure the Kubelet is new enough to support this.
+
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+There will be no feature gate. On a live cluster, admins can disable by not using the drop-in configuration directory and flag.
+
+
+###### Does enabling the feature change any default behavior?
+
+No, upgrading to a version of the Kubelet with this feature will not enable the Kubelet to be configured with the drop-in directory if no flag is specified.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No, only a rollback of the Kubelet can roll back the enablement.
+
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+This will be a Kubelet upgrade, which will re-enable the feature as expected.
```

Comment:
Same question about flag handling as above.

Reply:
yeah adding a blurb about the flag would be good too

Reply:
updated, thanks!